### PR TITLE
Dimagi utils spreadsheets

### PR DESCRIPTION
**Pull request by dannyroberts**
Move some spreadsheet utils from `dimagi.utils` to `corehq.util.spreadsheets`. I noticed that all of these spreadsheet utils were only being used by the commcare-hq repo, and were the last remaining things keeping both soil and couchexport necessarily submodules.

### CODE REVIEW

**Comment by dimagimon** [comment]
There are 10 errors:

* corehq/util/spreadsheets/excel.py, line 27 - E301 expected 1 blank line, found 0
* corehq/util/spreadsheets/excel.py, line 30 - E301 expected 1 blank line, found 0
* corehq/util/spreadsheets/excel.py, line 106 - W601 .has_key() is deprecated, use 'in'
* corehq/util/spreadsheets/excel.py, line 122 - W601 .has_key() is deprecated, use 'in'
* corehq/util/spreadsheets/excel.py, line 154 - W601 .has_key() is deprecated, use 'in'
* corehq/util/spreadsheets/excel.py, line 182 - E301 expected 1 blank line, found 0
* corehq/util/spreadsheets/export.py, line 6 - E301 expected 1 blank line, found 0
* corehq/util/tests/test_spreadsheets.py, line 7 - E301 expected 1 blank line, found 0
* corehq/util/tests/test_spreadsheets.py, line 17 - E501 line too long (118 > 115 characters)
* corehq/util/tests/test_spreadsheets.py, line 17 - E203 whitespace before ','

**Comment by dannyroberts** [comment]
The huge commit history is misleading; it's because I'm doing this as a (partial) subtree merge, in order to preserve git history on these files.

**Comment by dimagimon** [comment]
There are 10 errors:

* corehq/util/spreadsheets/excel.py, line 27 - E301 expected 1 blank line, found 0
* corehq/util/spreadsheets/excel.py, line 30 - E301 expected 1 blank line, found 0
* corehq/util/spreadsheets/excel.py, line 106 - W601 .has_key() is deprecated, use 'in'
* corehq/util/spreadsheets/excel.py, line 122 - W601 .has_key() is deprecated, use 'in'
* corehq/util/spreadsheets/excel.py, line 154 - W601 .has_key() is deprecated, use 'in'
* corehq/util/spreadsheets/excel.py, line 182 - E301 expected 1 blank line, found 0
* corehq/util/spreadsheets/export.py, line 6 - E301 expected 1 blank line, found 0
* corehq/util/tests/test_spreadsheets.py, line 7 - E301 expected 1 blank line, found 0
* corehq/util/tests/test_spreadsheets.py, line 17 - E501 line too long (118 > 115 characters)
* corehq/util/tests/test_spreadsheets.py, line 17 - E203 whitespace before ','

File: corehq/util/spreadsheets/excel.py [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+            value = value.strip()
+        # try dict
+        try:
+            field, subfield = field.split(':')
+        except Exception:
+            pass
+        else:
+            field = field.strip()
+
+            if not field in obj:
```
Comment (dimagimon) [inline corehq/util/spreadsheets/excel.py]: E713 test for membership should be 'not in'

File: corehq/util/spreadsheets/excel.py [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+        try:
+            field, _ = field.split()
+        except Exception:
+            pass
+        else:
+            dud = {}
+            cls.set_field_value(dud, field, value)
+            (field, value), = dud.items()
+
+            if not field in obj:
```
Comment (dimagimon) [inline corehq/util/spreadsheets/excel.py]: E713 test for membership should be 'not in'

File: corehq/util/spreadsheets/excel.py (hunk 1 above)
Comment (dimagimon) [inline corehq/util/spreadsheets/excel.py]: E713 test for membership should be 'not in'

File: corehq/util/spreadsheets/excel.py (hunk 2 above)
Comment (dimagimon) [inline corehq/util/spreadsheets/excel.py]: E713 test for membership should be 'not in'

**Pull request merged**